src/Database/Type/DateType.php

```diff
@@ -80,7 +81,9 @@ public function useMutable()
     public function marshal($value): ?DateTimeInterface
     {
         $date = parent::marshal($value);
-        if ($date instanceof DateTime) {
+        if ($date && !$date instanceof I18nDateTimeInterface) {
```

Comment:
This is the issue I noticed. When I18n isn't available (something we don't test), only mutable dates would be cleared properly.